src/Compilers/CSharp/Portable/Symbols/Synthesized/InjectedNonNullTypesAttributeSymbol.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Makes the System.Runtime.CompilerServices.NonNullTypesAttribute available in every compilation.
+    /// </summary>
+    internal sealed class InjectedNonNullTypesAttributeSymbol : InjectedAttributeSymbol
+    {
+        private ImmutableArray<CSharpAttributeData> _lazyCustomAttributes;
+        private SymbolCompletionState _state;
+
+        private InjectedNonNullTypesAttributeSymbol(
+            AttributeDescription description,
+            NamespaceSymbol containingNamespace,
+            CSharpCompilation compilation,
+            Func<CSharpCompilation, NamedTypeSymbol, DiagnosticBag, ImmutableArray<MethodSymbol>> getConstructors)
+            : base(description, containingNamespace, compilation, getConstructors)
+        {
+        }
+
+        public static InjectedNonNullTypesAttributeSymbol Create(NamespaceSymbol containingNamespace)
+        {
+            return new InjectedNonNullTypesAttributeSymbol(AttributeDescription.NonNullTypesAttribute, containingNamespace, containingNamespace.DeclaringCompilation, makeNonNullTypesAttributeConstructor);
+
+            ImmutableArray<MethodSymbol> makeNonNullTypesAttributeConstructor(CSharpCompilation compilation, NamedTypeSymbol containingType, DiagnosticBag diagnostics)
+            {
+                var boolType = compilation.GetSpecialType(SpecialType.System_Boolean);
+
+                Binder.ReportUseSiteDiagnostics(boolType, diagnostics, Location.None);
+
+                var boolWithAnnotations = TypeSymbolWithAnnotations.Create(boolType);
+                // PROTOTYPE constructor should save the parameter into a field (for users of reflection)
+                return ImmutableArray.Create<MethodSymbol>(
+                    new NonNullTypesAttributeConstructorSymbol(
+                        containingType,
+                        m => ImmutableArray.Create(SynthesizedParameterSymbol.Create(m, boolWithAnnotations, 0, ConstantValue.True, name: "flag"))));
+            }
+        }
+
+        internal override bool RequiresCompletion
+            => true;
```

Comment:
> => true; [](start = 11, length = 9)

I think use of completion for injected symbols is an overkill. Its purpose is to make sure all declaration diagnostics is reported into Compilation-wide bag, but these symbols never do that.

Reply:
`ForceComplete` seems a standard mechanism to enforce that all parts are pull on. In the case of injected symbols, some parts were not naturally getting pulled on (I'll confirm and find the specific scenario), so this seems a reasonable approach.
The fact that the diagnostics happen to be captured for deferred reporting (only if the injected types end up being used) doesn't really change that. 

The alternative seems to be to add some custom logic to pull on those parts when emitting, but that feels more awkward and ad-hoc.